### ISSUE TITLE
Integrate with google/oss-fuzz

### ISSUE THREAD

**Issue by silvergasp** [jarro2783/cxxopts#387]
Hey I'd like to suggest adding cxxopts to google/oss-fuzz. If you aren't familiar with fuzz testing, here is a bit of a run down ([from Wikipedia](https://en.wikipedia.org/wiki/Fuzzing));

> In [programming](https://en.wikipedia.org/wiki/Computer_programming) and [software development](https://en.wikipedia.org/wiki/Software_development), fuzzing or fuzz testing is an automated [software testing](https://en.wikipedia.org/wiki/Software_testing) technique that involves providing invalid, unexpected, or [random data](https://en.wikipedia.org/wiki/Random_data) as inputs to a [computer program](https://en.wikipedia.org/wiki/Computer_program). The program is then monitored for exceptions such as [crashes](https://en.wikipedia.org/wiki/Crash_(computing)), failing built-in code [assertions](https://en.wikipedia.org/wiki/Assertion_(software_development)), or potential [memory leaks](https://en.wikipedia.org/wiki/Memory_leak).

Google offers a free continuous fuzzing service called OSS-fuzz. If cxxopts is integrated into oss-fuzz, the fuzz tests under cxxopts will be built and then run once a day, to search for bugs and vulnerabilities in cxxopts. This service can be integrated with the CI for cxxopts, so that the fuzz tests are run for 10min or so for every pull request, preventing buggy code from being merged.

I've opened up a pull request to add a basic fuzz-testing harness here https://github.com/jarro2783/cxxopts/pull/386. If you are keen on adding cxxopts to oss-fuzz I'd be happy to champion the integration :)

**Comment by jarro2783** [comment]
That looks interesting. I'll take a look at your PR.

**Comment by silvergasp** [comment]
Hey @jarro2783, I've had some time to start integrating cxxopts with OSS-fuzz. I've got a draft PR here https://github.com/google/oss-fuzz/pull/9679, all I need from you is an email address associated with a google account to add to the [primary_contact](https://github.com/google/oss-fuzz/pull/9679/files#diff-65da3c099e817bcbe5fc03abe6456062021644d0fbe50dfe476b7d83e82c1fbeR3). You can read about why a google account is required [here](https://google.github.io/oss-fuzz/faq/#why-do-you-require-a-google-account-for-authentication).

As an additional step, if you are happy to go ahead with integrating with OSS-fuzz I'll need you to comment on the PR  mentioning that you approve of the request to join the program :)

**Comment by jarro2783** [comment]
Yes you can use the email address that I use in my commits.

**Comment by silvergasp** [comment]
> Yes you can use the email address that I use in my commits.

[Done](https://github.com/google/oss-fuzz/pull/9679/files#diff-65da3c099e817bcbe5fc03abe6456062021644d0fbe50dfe476b7d83e82c1fbeR3)

**Comment by silvergasp** [comment]
So this is now integrated into oss-fuzz. I've just opened up a PR #392 to run the fuzzer for 10min or so to catch shallow bugs in pull requests. This is in addition to the once per day fuzzing in oss-fuzz.

On a different note, are you receiving notifications from OSS-fuzz about failing coverage collection? I won't receive these notifications I'm not listed as the primary_contact. But I checked and it looks like there might be some issue with the code-coverage collection. I might give it a couple more days, I think code coverage is calculated every other day, so it might just be a timing thing.
